### PR TITLE
Handle top level errors coming from Turbopack entrypoints subscription

### PR DESCRIPTION
**Pull request by timneutkens**
## What?

Adds handling for issues that are emitted during the entrypoints subscription, before the changes in this PR any issue (i.e. compiler errors) that was emitted during the collection of which entrypoints exist in the application were ignored.

Closes NEXT-2591

### CODE REVIEW

**Comment by timneutkens** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/vercel/next.js/62528?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#62528** <a href="https://app.graphite.dev/github/pr/vercel/next.js/62528?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#62526** <a href="https://app.graphite.dev/github/pr/vercel/next.js/62526?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#62524** <a href="https://app.graphite.dev/github/pr/vercel/next.js/62524?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `canary`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @timneutkens and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Comment by ijjk** [comment]
## Failing test suites 
Commit: 97ad90459cfb8ecca55f75ba3024f5f7f05c1dd7

`pnpm test-start test/e2e/app-dir/navigation/navigation.test.ts`
- app dir - navigation > query string > should handle unicode search params

<details>
<summary>Expand output</summary>

  ● app dir - navigation › query string › should handle unicode search params

    request.response: Target page, context or browser has been closed

      37 |         const browser = await next.browser('/search-params?name=名')
      38 |         async function requestHandler(req: Request) {
    > 39 |           const res = await req.response()
         |                                 ^
      40 |           if (!res) return
      41 |
      42 |           requests.push({

      at response (e2e/app-dir/navigation/navigation.test.ts:39:33)
      at cb (lib/browsers/playwright.ts:234:51)
      at Page.forEach (lib/browsers/playwright.ts:234:35)

</details>


Read more about building and testing Next.js in [contributing.md](https://github.com/vercel/next.js/blob/canary/contributing.md).

**Comment by ijjk** [comment]
# Stats from current PR

<details open>
<summary><strong>Default Build</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| buildDuration | 14s | 14s | N/A |
| buildDurationCached | 7.7s | 6.5s | N/A |
| nodeModulesSize | 197 MB | 197 MB | ⚠️ +8.75 kB |
| nextStartRea..uration (ms) | 432ms | 418ms | N/A |

</details>

<details>
<summary><strong>Client Bundles (main, webpack)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| 1068-HASH.js gzip | 30.3 kB | 30.3 kB | N/A |
| 3f784ff6-HASH.js gzip | 53.7 kB | 53.7 kB | N/A |
| 4944-HASH.js gzip | 5.04 kB | 5.03 kB | N/A |
| 8423.HASH.js gzip | 181 B | 181 B | ✓ |
| framework-HASH.js gzip | 45.2 kB | 45.2 kB | ✓ |
| main-app-HASH.js gzip | 241 B | 242 B | N/A |
| main-HASH.js gzip | 32.2 kB | 32.1 kB | N/A |
| webpack-HASH.js gzip | 1.7 kB | 1.7 kB | ✓ |
| Overall change | 47.1 kB | 47.1 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| polyfills-HASH.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| _app-HASH.js gzip | 196 B | 196 B | ✓ |
| _error-HASH.js gzip | 184 B | 183 B | N/A |
| amp-HASH.js gzip | 503 B | 504 B | N/A |
| css-HASH.js gzip | 323 B | 324 B | N/A |
| dynamic-HASH.js gzip | 2.5 kB | 2.51 kB | N/A |
| edge-ssr-HASH.js gzip | 258 B | 259 B | N/A |
| head-HASH.js gzip | 353 B | 351 B | N/A |
| hooks-HASH.js gzip | 370 B | 370 B | ✓ |
| image-HASH.js gzip | 4.21 kB | 4.2 kB | N/A |
| index-HASH.js gzip | 259 B | 259 B | ✓ |
| link-HASH.js gzip | 2.68 kB | 2.67 kB | N/A |
| routerDirect..HASH.js gzip | 313 B | 314 B | N/A |
| script-HASH.js gzip | 386 B | 385 B | N/A |
| withRouter-HASH.js gzip | 309 B | 311 B | N/A |
| 1afbb74e6ecf..834.css gzip | 106 B | 106 B | ✓ |
| Overall change | 931 B | 931 B | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| _buildManifest.js gzip | 485 B | 484 B | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| index.html gzip | 528 B | 528 B | ✓ |
| link.html gzip | 540 B | 540 B | ✓ |
| withRouter.html gzip | 523 B | 522 B | N/A |
| Overall change | 1.07 kB | 1.07 kB | ✓ |

</details>

<details>
<summary><strong>Edge SSR bundle Size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| edge-ssr.js gzip | 94.9 kB | 94.9 kB | N/A |
| page.js gzip | 3.06 kB | 3.05 kB | N/A |
| Overall change | 0 B | 0 B | ✓ |

</details>

<details>
<summary><strong>Middleware size</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| middleware-b..fest.js gzip | 627 B | 627 B | ✓ |
| middleware-r..fest.js gzip | 151 B | 151 B | ✓ |
| middleware.js gzip | 25.4 kB | 25.4 kB | ✓ |
| edge-runtime..pack.js gzip | 839 B | 839 B | ✓ |
| Overall change | 27.1 kB | 27.1 kB | ✓ |

</details>

<details>
<summary><strong>Next Runtimes</strong></summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| app-page-exp...dev.js gzip | 170 kB | 170 kB | ✓ |
| app-page-exp..prod.js gzip | 96.5 kB | 96.5 kB | ✓ |
| app-page-tur..prod.js gzip | 98.3 kB | 98.3 kB | ✓ |
| app-page-tur..prod.js gzip | 92.8 kB | 92.8 kB | ✓ |
| app-page.run...dev.js gzip | 151 kB | 151 kB | ✓ |
| app-page.run..prod.js gzip | 91.3 kB | 91.3 kB | ✓ |
| app-route-ex...dev.js gzip | 22 kB | 22 kB | ✓ |
| app-route-ex..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 15 kB | 15 kB | ✓ |
| app-route-tu..prod.js gzip | 14.7 kB | 14.7 kB | ✓ |
| app-route.ru...dev.js gzip | 21.7 kB | 21.7 kB | ✓ |
| app-route.ru..prod.js gzip | 14.7 kB | 14.7 kB | ✓ |
| pages-api-tu..prod.js gzip | 9.5 kB | 9.5 kB | ✓ |
| pages-api.ru...dev.js gzip | 9.78 kB | 9.78 kB | ✓ |
| pages-api.ru..prod.js gzip | 9.5 kB | 9.5 kB | ✓ |
| pages-turbo...prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| pages.runtim...dev.js gzip | 23 kB | 23 kB | ✓ |
| pages.runtim..prod.js gzip | 22.3 kB | 22.3 kB | ✓ |
| server.runti..prod.js gzip | 50.5 kB | 50.5 kB | ✓ |
| Overall change | 950 kB | 950 kB | ✓ |

</details>

<details>
<summary><strong>build cache</strong> Overall increase ⚠️</summary>

|  | vercel/next.js canary  | vercel/next.js 02-26-Handle_top_level_errors_coming_from_Turbopack_entrypoints_subscription | Change |
| - | - | - | - |
| 0.pack gzip | 1.55 MB | 1.56 MB | ⚠️ +276 B |
| index.pack gzip | 104 kB | 105 kB | ⚠️ +434 B |
| Overall change | 1.66 MB | 1.66 MB | ⚠️ +710 B |

</details>

</details>
Commit: 97ad90459cfb8ecca55f75ba3024f5f7f05c1dd7

File: packages/next/src/server/dev/hot-reloader-turbopack.ts [hunk 1]
```diff
@@ -262,6 +269,10 @@ export async function createHotReloaderTurbopack(
     for (const client of clients) {
       const state = clientStates.get(client)
       if (!state) continue
+      if (currentTopLevelIssues.size > 0) {
+        // During compilation errors we want to delay the HMR events until errors are fixed
+        return
+      }
```
Comment (sokra) [inline packages/next/src/server/dev/hot-reloader-turbopack.ts:275]: This isn't needed. You already checked above

**Review by sokra — APPROVED** [review]
Do you also want to handle them in `build`

**Comment by timneutkens** [comment]
Added handling for build 🙌

**Comment by timneutkens** [comment]
This is blocked on #62585 as not-found handling causes a conflict

**Pull request merged**